src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -365,6 +365,10 @@ case object JUMPI extends OpCode(0x57, 2, 0) {
 }
 
 case object JUMPDEST extends OpCode(0x5b, 0, 0) {
+  // TODO To correctly implement this opcode EVM has to scan whole bytecode of a program
+  // to find all valid JUMP destinations.
+  // We will also have to scan for SSTORE and SSLOAD opcodes and if they are not found
+  // then there's no sense in loading Storage of a transaction from MPT
```

Comment:
1) The comment is misplaced because JUMP and JUMPI have to be validated prior to reaching a JUMPDEST (and those instructions already have the comments).

2) SSTORE and SSLOAD wouldn't be enough: http://ethereum.stackexchange.com/questions/3667/difference-between-call-callcode-and-delegatecall 
Anyway, I would not force ourselves onto this optimisation yet. It may actually be uncalled for if we decide to fetch stored values on the fly.